lib/compress.js

```diff
@@ -2492,6 +2494,7 @@ merge(Compressor.prototype, {
             }
         }
         self = self.lift_sequences(compressor);
+        if (!(self instanceof AST_Binary)) return self;
```

Comment:
Not directly related to `AST_Seq` per se, but this is what caught me out when working on #1427 - feels like a miracle that all of the code below, which implicitly assumes `self` remains to be `AST_Binary`, breaks in any shape or form.